### PR TITLE
[FIX] account: cannot make branch with generic coa

### PR DESCRIPTION
**Pull request by ande-odoo**
Current behaviour:
---
When your current company is South Korean,
The chart of account is generic.
If you try to create a South Korean company
branch, you'll get an error message:
The Currency of a subsidiary must be the same as its root company.

Steps to reproduce:
---
1. Install account_accountant
2. Go to Settings
3. Go to Users & Companies > Companies
4. Create a new company
5. Country: South Korea
6. Currency: KRW
7. Switch into the new company (top right)
8. Go to Accounting
9. Go to Configuration > Settings
10. Fiscal Localization: Generic Chart Template
11. Save the changes
12. Fiscal Country: South Korea
13. Main currency: KRW
14. Save the changes
15. Go to Settings
16. Go to Users & Companies > Companies
17. Select the Korean company
18. Create a new branch
19. Country: South Korea
20. The Currency of a subsidiary must be the same as it's root company.

Cause of the issue:
---
The generic chart of account's currency is USD

opw-3537346

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by agr-odoo — APPROVED** [review]
Looks ok to me

**Comment by Levizar** [comment]
🤔🤔🤔
image

File: addons/account/models/chart_template.py [hunk 1]
```diff
@@ -362,7 +362,9 @@ def _pre_load_data(self, template_code, company, template_data, data):
         e.g. the account codes' width must be standardized to the code_digits applied.
         The fiscal country code must be put in place before taxes are generated.
         """
-        if 'account_fiscal_country_id' in data['res.company'][company.id]:
+        if company.parent_id:
+            fiscal_country = company.parent_id.account_fiscal_country_id
+        elif 'account_fiscal_country_id' in data['res.company'][company.id]:
```
Comment (Levizar) [inline addons/account/models/chart_template.py:367]: I think this isn't the right fix.
Here, we are forcing the branch to have the same fiscal country while we just want it to have the same currency, which is set below on line 382.

I think it's there that we should use the same condition and assign the parent company currency.

File: addons/account/tests/test_company_branch.py [hunk 2]
```diff
@@ -206,3 +206,26 @@ def test_change_record_company(self):
...
+    def test_create_branch_generic_coa(self):
+        root_company = self.env['res.company'].create({
+            'name': 'Korean Company',
+            'country_id': self.env.ref('base.kr').id,
+        })
+        chart_template = self.env['account.chart.template']._guess_chart_template(root_company.country_id)
+        self.env['account.chart.template'].try_loading(chart_template, company=root_company, install_demo=False)
+        root_company.write({
+            'currency_id': self.env.ref('base.KRW').id,
+            'account_fiscal_country_id': root_company.country_id.id,
```
[the remark this answers is not in the document]
Comment (ande-odoo) [reply addons/account/tests/test_company_branch.py]: No because try_loading forces currency_id to USD and account_fiscal_country_id to the United States

**Review by Levizar — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**